subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskToolchainIntegrationTest.groovy

```diff
@@ -77,38 +179,85 @@ class TestTaskToolchainIntegrationTest extends AbstractIntegrationSpec {
+    private TestFile configureProjectWithoutJavaBasePlugin(JavaVersion compileJavaVersion) {
+        buildFile << """
+            plugins {
+                id 'jvm-toolchains'
+            }
 
-        when:
-        result = executer
-            .withArguments("-Porg.gradle.java.installations.paths=" + someJdk.javaHome.absolutePath, "--info")
-            .withTasks("test")
-            .run()
+            ${mavenCentralRepository()}
 
-        then:
-        outputContains("Tests running with ${someJdk.javaHome.absolutePath}")
-        noExceptionThrown()
+            configurations {
+                testImplementation
+            }
+
+            dependencies {
+                testImplementation 'junit:junit:4.13'
+            }
+
+            def buildDir = project.layout.buildDirectory
+
+            task compileJava(type: JavaCompile) {
+                classpath = configurations.testImplementation
+                source = project.layout.files("src/test/java")
+                destinationDirectory = buildDir.dir("classes/java/test")
+                sourceCompatibility = "${compileJavaVersion}"
+                targetCompatibility = "${compileJavaVersion}"
+            }
+
+            task test(type: Test) {
+                testClassesDirs = buildDir.files("classes/java/test")
+                classpath = files(configurations.testImplementation, buildDir.dir("classes/java/test"))
+                binaryResultsDirectory.set(buildDir.dir("test-results"))
+                reports.junitXml.required.set(false)
+                reports.html.outputLocation.set(buildDir.dir("test-results"))
+            }
+
+            test.dependsOn compileJava
+        """
     }
 
-    private static String testClass(String className) {
-        return """
-            import org.junit.*;
+    private TestFile configureJavaExtension(Jvm jdk) {
+        buildFile << """
+            java {
+                toolchain {
+                    languageVersion = JavaLanguageVersion.of(${jdk.javaVersion.majorVersion})
+                }
+            }
+        """
+    }
 
-            public class $className {
-               @Test
-               public void test() {
-                  System.out.println("Tests running with " + System.getProperty("java.home"));
-                  Assert.assertEquals(1,1);
-               }
+    private TestFile configureExecutable(Jvm jdk) {
+        buildFile << """
+            test {
+                executable = "${TextUtil.normaliseFileSeparators(jdk.javaExecutable.absolutePath)}"
             }
-        """.stripIndent()
+        """
+    }
+
+    private TestFile configureLauncher(Jvm jdk) {
+        buildFile << """
+            test {
+                javaLauncher = javaToolchains.launcherFor {
+                    languageVersion = JavaLanguageVersion.of(${jdk.javaVersion.majorVersion})
+                }
+            }
+        """
+    }
+
+    private withInstallations(Jvm... jvm) {
```

Comment:
Is this something which should be extracted to a general `JavaToolchainFixture` trait?

Reply:
I was thinking about this for a while. Now that you mention it, we should definitely do it.